processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/BadRecordsLogger.java

```diff
@@ -87,22 +87,25 @@
 
   private boolean badRecordConvertNullDisable;
 
+  private boolean isDataLoadFail;
+
   // private final Object syncObject =new Object();
 
   public BadRecordsLogger(String key, String fileName, String storePath,
       boolean badRecordsLogRedirect, boolean badRecordLoggerEnable,
-      boolean badRecordConvertNullDisable) {
+      boolean badRecordConvertNullDisable, boolean isDataLoadFail) {
```

Comment:
Please set  this bad records related properties to a bad record type VO object and pass the VO object as argument for better maintanability, else the arguments in constructor will be increasing in future.